nowcasting_dataset/data_sources/data_source.py

```diff
@@ -456,3 +456,112 @@ def open(self) -> None:
 
     def _open_data(self) -> xr.DataArray:
         raise NotImplementedError()
+
+
+@dataclass
+class DerivedDataSource(DataSource):
+    """
+    Base class for data sources derived from other data sources
+    """
+
+    history_minutes: int = 0
+    forecast_minutes: int = 0
+
+    def datetime_index(self):
+        """The datetime index of this datasource"""
+        return NotImplementedError(
+            "DerivedDataSources only use other, pre-computed batches, so no datetime_index is "
+            "needed"
+        )
+
+    def create_batches(
+        self,
+        batch_path: Path,
+        total_number_batches: int,
+        idx_of_first_batch: int,
+        dst_path: Path,
+        local_temp_path: Path,
+        upload_every_n_batches: int,
+        **kwargs,
+    ) -> None:
+        """Create multiple batches and save them to disk.
+
+        Safe to call from worker processes.
+
+        Args:
+          batch_path: Path to where the netcdf batches are stored
+          total_number_batches: The total number of batches to make
+          idx_of_first_batch: The batch number of the first batch to create.
+          dst_path: The final destination path for the batches.  Must exist.
+          local_temp_path: The local temporary path.  This is only required when dst_path is a
+            cloud storage bucket, so files must first be created on the VM's local disk in temp_path
+            and then uploaded to dst_path every upload_every_n_batches. Must exist. Will be emptied.
+          upload_every_n_batches: Upload the contents of temp_path to dst_path after this number
+            of batches have been created.  If 0 then will write directly to dst_path.
+        """
+        # Sanity checks:
+        assert idx_of_first_batch >= 0
+        assert upload_every_n_batches >= 0
+        assert total_number_batches >= 0
+
+        self.open()
```

Comment:
is this copied from Datasource, can re remov any duplicated code?

Reply:
Yeah, its close to the original data source, so I can probably reduce it a bit. There is a same thing in Manager. There are some differences in terms of this loads netCDF batches and does not use any of the t0 datetimes, x,y centers, etc.